bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -161,6 +215,44 @@ private void write(Page page, XMLStreamWriter stream, ResourceResolver resolver)
         stream.writeEndElement();
     }
 
+    private void writeAsset(Asset asset, XMLStreamWriter stream, ResourceResolver resolver) throws XMLStreamException {
+        stream.writeStartElement(NS, "url");
+
+
+        String loc = externalizer.externalLink(resolver, externalizerDomain, asset.getPath());
+        writeElement(stream, "loc", loc);
+
+        if (includeLastModified) {
+            long lastModified = asset.getLastModified();
+            if (lastModified > 0) {
+                writeElement(stream, "lastmod", DATE_FORMAT.format(lastModified));
+            }
+        }
+
+        Resource contentResource = asset.adaptTo(Resource.class).getChild("jcr:content");
+        if (contentResource != null) {
+            final ValueMap properties = contentResource.getValueMap();
+            writeFirstPropertyValue(stream, "changefreq", changefreqProperties, properties);
+            writeFirstPropertyValue(stream, "priority", priorityProperties, properties);
+        }
+
+        stream.writeEndElement();
+    }
+
+    private void writeAssets(final XMLStreamWriter stream, final Resource assetFolder, final ResourceResolver resolver)
+            throws XMLStreamException {
+        for (Iterator<Resource> children = assetFolder.listChildren(); children.hasNext();) {
+            Resource assetFolderChild = children.next();
+            if (assetFolderChild.isResourceType("dam:Asset")) {
+                Asset asset = assetFolderChild.adaptTo(Asset.class);
+
+                if (damAssetTypes.contains(asset.getMimeType())) {
```

Comment:
WDYT about making this a bit more flexible so I could specify something like image/* (for all image mimetypes) ? I guess we could always add that later as it would be totally backwards compatible.